mlir/lib/Dialect/Rock/Transforms/AffixTuningParameters.cpp

```diff
@@ -58,6 +58,13 @@ void AffixTuningParameters::runOnOperation() {
 
   func.walk(
       [&](RockGemmWrapperInterface op) { affixTuningParametersImpl(op); });
+  func.walk([&](ReduceOp op) {
+    func::FuncOp funcOp = getOperation();
+    if (!funcOp->hasAttr("block_size")) {
+      funcOp->setAttr("block_size", op.getBlockSizeAttr());
+      funcOp->setAttr("grid_size", op.getGridSizeAttr());
```

Comment:
what happens if these are already set on the func, how do you resolve?

Reply:
its based on precedence and right now gemm like ops takes precedence (if they are present and that scenario represent a fusion case).  What I've codified here is the walk order by inserting the reduction op walker after gemm-like op walker.